### PR TITLE
Add support for SSL client certs

### PR DESCRIPTION
**Pull request by nmelo**

### CODE REVIEW

File: boom.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ import (
 	"runtime"
 	"strings"
 
+	"crypto/tls"
```
Comment (rakyll) [inline boom.go]: Stdlib imports need to be grouped together.

Remove the line before crypto/tls, add a line after path/filepath.

File: boom.go [hunk 2]
```diff
@@ -180,6 +197,7 @@ func main() {
 		DisableKeepAlives:  *disableKeepAlives,
 		ProxyAddr:          proxyURL,
 		Output:             *output,
+		Certificate: 		tlsCert,
```
Comment (rakyll) [inline boom.go]: gofmt

File: boom.go [hunk 3]
```diff
@@ -160,6 +168,15 @@ func main() {
 		}
 	}
 
+	var tlsCert tls.Certificate
+	if *cert != "" && *key != "" {
+		var err error
+		tlsCert, err = tls.LoadX509KeyPair(filepath.Join("", *cert), filepath.Join("", *key))
```
Comment (rakyll) [inline boom.go]: s/filepath.Join/filepath.Clean?

File: boom.go [hunk 4]
```diff
@@ -60,6 +62,9 @@ var (
 	t    = flag.Int("t", 0, "")
 	cpus = flag.Int("cpus", runtime.GOMAXPROCS(-1), "")
 
+	cert = flag.String("cert", "", "")
+	key  = flag.String("key", "", "")
```
Comment (rakyll) [inline boom.go]: key is really generic, certkey or something more specific?

**Comment by rakyll** [comment]
I am just wondering whether we should make them env variables. I can see that the number of flags required for SSL can grow easily.

File: boom.go (hunk 4 above)
Comment (nmelo) [inline boom.go]: curl uses --cert and --key but maybe --privkey?

**Comment by nmelo** [comment]
@rakyll Agree. I can make a separate PR with the cert and key as env vars as well as CLI flags

**Comment by rakyll** [comment]
The project has moved to https://github.com/rakyll/hey. Closing the PRs on this legacy repo.

**Pull request closed without merging**